specification/ai/Azure.AI.Agents/tools/tool_resources.tsp

```diff
@@ -129,137 +129,6 @@ model AISearchIndexResource {
    *  filter string for search resource.
    */
   filter?: string;
-
-  // TODO: Jose: Change to union type
-  /**
-   * Index asset id for search resource.
-   */
-  @encodedName("application/json", "index_asset_id")
-  indexAssetId?: string;
```

Comment:
Is this not needed?

Reply:
@glharper please check this also

Reply:
Thanks for confirming, @ninghu we need to add this back

Reply:
from AK: 

remove it from this (because this typespec model has 3 params)
fast follow with a new model that has only asset id
include a union[connection, asset] as the config on the tool